### ISSUE TITLE
Nice work.

### ISSUE THREAD

**Issue by davidcoles** [GJSBRT/hashed-weighted-round-robin#1]
This all seems pretty sound, although I'm not sure round-robin is the best description - more like weighted-hash?

The problem that you could run into is when you need to add or remove backends from the pool (eg. when reconfigured or a backend fails health checks). This would change to the length of the array, which makes a drastic change to selections of backends, and would break the majority of connections.

This can be avoided by using a [consistent hashing](https://en.wikipedia.org/wiki/Consistent_hashing) algorithm which results in only losing around 1/n-th of the connections when the pool changes. Rendezvous hasing or Maglev are two examples - I found the worked example from page 32 onwards in this description of Google's internal project to be quite easy to follow:

https://www.usenix.org/sites/default/files/conference/protected-files/nsdi16_slides_eisenbud.pdf

**Comment by GJSBRT** [comment]
Hi, Thanks!

I agree that the name is not correct. Naming things is hard ;p.

I did realize that changing the list of destinations would cause connections to break, however I did not think much of it as it is possible to calculate the new mapping, but now that I think a bit more about it, this is only useful if a planned change happens and not when a health check fails like you mentioned.

Thanks for letting me know about these algorithms. It seems I have much more reading to do :).
